### DISCUSSION TITLE
Shared backDrawRodData between multiple rods

### DISCUSSION THREAD

**Discussion by ospaarmann** [imaNNeo/fl_chart#1708] [General]
First of all: Thanks for the great library.

I try to implement a graph where I have two rods for each day and a shared backgorund. This seems to not be possible. Maybe I'm missing something. If not: That is in my opinion a very common design and would be a great addition to the library.

This is how it should look like:

<img width="508" alt="Screenshot 2024-07-11 at 13 14 22" src="https://github.com/imaNNeo/fl_chart/assets/308086/9db178b5-9e91-439b-99f9-2a2fe02f4060">

This is what I managed to do:

<img width="444" alt="Screenshot 2024-07-11 at 13 14 58" src="https://github.com/imaNNeo/fl_chart/assets/308086/cf4aa5f7-21c9-41be-8010-63a784e42f77">

The code that does not achieve it... If I add `backDrawRodData` to both rods, I just get two distinct backgrounds.

```dart
  List<BarChartGroupData> _barGroupsData(
      List<double> firstScoresByWeekday, List<double> secondScoresByWeekday) {
    return List.generate(7, (index) {
      final double firstScore = firstScoresByWeekday[index];
      final double secondScore = secondScoresByWeekday[index];

      return BarChartGroupData(
        x: index,
        barRods: [
          BarChartRodData(
            toY: firstScore * 10, 
            color: AppColors.blue900,
            width: 12, 
            borderRadius: const BorderRadius.all(Radius.circular(20)),
            backDrawRodData: BackgroundBarChartRodData(
              show: true,
              toY: 10,
              color: AppColors.blue100,
            ),
          ),
          BarChartRodData(
              toY:
                  secondScore * 10, 
              color:
                  AppColors.blue300, 
              width: 12, 
              borderRadius: const BorderRadius.all(Radius.circular(20)),
              backDrawRodData: BackgroundBarChartRodData(
                show: true,
                toY: 10,
                color: AppColors.blue100,
              )),
        ],
        // Optional: Add some space between the rods
        barsSpace: 0,
      );
    });
  }
```

**Comment by ospaarmann** [comment]
To add to this: If it was possible to set the background on BarChartGroupData, that would fix the issue. If this is something people want I would offer to raise a PR.

**Reply by imaNNeo** [reply]
Yes exactly, we can add a property in the BarChartGroupData that accepts a background. But I'm not sure what type can we use, maybe a [BoxDecoration](https://api.flutter.dev/flutter/painting/BoxDecoration-class.html) and paint it like this:

https://github.com/flutter/flutter/blob/14f9b568fc57c14d67b5b40bd48b7e6cb780319c/packages/flutter/lib/src/rendering/proxy_box.dart#L2313-L2321

**Reply by imaNNeo** [reply]
But you definitely can work on it.
